graylog2-web-interface/src/components/extractors/ExtractorsList.jsx

```diff
@@ -15,92 +15,94 @@
  * <http://www.mongodb.com/licensing/server-side-public-license>.
  */
 import PropTypes from 'prop-types';
-import React from 'react';
-import createReactClass from 'create-react-class';
-import Reflux from 'reflux';
+import * as React from 'react';
+import { useEffect, useRef, useState } from 'react';
 
-import { defaultCompare as naturalSort } from 'logic/DefaultCompare';
 import { Row, Col, Button } from 'components/bootstrap';
 import Spinner from 'components/common/Spinner';
 import AddExtractorWizard from 'components/extractors/AddExtractorWizard';
 import EntityList from 'components/common/EntityList';
-import { ExtractorsActions, ExtractorsStore } from 'stores/extractors/ExtractorsStore';
+import { ExtractorsActions } from 'stores/extractors/ExtractorsStore';
 
 import ExtractorsListItem from './ExtractorsListItem';
 import ExtractorsSortModal from './ExtractorSortModal';
 
-const ExtractorsList = createReactClass({
-  displayName: 'ExtractorsList',
+const fetchExtractors = (inputId, callback) => {
+  ExtractorsActions.list.triggerPromise(inputId).then((data) => callback(data?.extractors));
+};
 
-  propTypes: {
-    input: PropTypes.object.isRequired,
-    node: PropTypes.object.isRequired,
-  },
+const ExtractorsList = ({ input, node }) => {
+  const [extractors, setExtractors] = useState(null);
+  const sortModal = useRef(null);
 
-  mixins: [Reflux.connect(ExtractorsStore), Reflux.ListenerMethods],
+  useEffect(() => {
+    fetchExtractors(input.id, setExtractors);
+  }, [input.id]);
 
-  componentDidMount() {
-    ExtractorsActions.list.triggerPromise(this.props.input.id);
-  },
-
-  _formatExtractor(extractor) {
+  const _formatExtractor = (extractor) => {
     return (
       <ExtractorsListItem key={extractor.id}
                           extractor={extractor}
-                          inputId={this.props.input.id}
-                          nodeId={this.props.node.node_id} />
+                          inputId={input.id}
+                          nodeId={node[0]?.node_id || ''} />
```

Comment:
The edit link is broken because the node is an object, not an array. I think this line should be as bellow:
```suggestion
                          nodeId={node?.node_id} />
```